src/Chainweb/Rosetta/RestAPI/Server.hs

```diff
@@ -139,7 +139,7 @@ constructionSubmitH v ms (ConstructionSubmitReq net@(NetworkId _ _ msni) tx) =
         SubNetworkId n _ <- msni ?? RosettaChainUnspecified
         cmd <- command tx ?? RosettaUnparsableTx
         validated <- hoistEither . first (const RosettaInvalidTx) $ validateCommand cmd
-        mp <- (readMaybe (T.unpack n) >>= flip lookup ms) ?? RosettaInvalidChain
```

Comment:
(related to above comment) lines 139 and lines 142 perform the same checks as the validate network function.